webpack/assets/javascripts/react_app/components/common/DateTimePicker/DateComponents/DateInput.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { addMonths } from './helpers';
+import MonthView from './MonthView';
+import YearView from './YearView';
+import DecadeView from './DecadeView';
+
+class DateInput extends React.Component {
+  state = {
+    selectedDate: new Date(this.props.date),
+    date: new Date(this.props.date),
+    typeOfDateInput: this.props.typeOfDateInput,
+  };
+  static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
`getDerivedStateFromProps` is running before each rendering,  according to react [doc](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) it should only be used in rare cases, consider using alternative

Reply:
Changed to componentDidUpdate, thanks!